### PR TITLE
feat: PHC-4775 Add Open Thread Screen Logic

### PR DESCRIPTION
**Pull request by sternetj**
## Changes

  - Wires up the discussion screen to open the thread screen when pressed
  - Fixes a path issue
  - Fixes the back arrow for the header (no more X)

## Screenshots

https://github.com/lifeomic/react-native-sdk/assets/2295908/f2ec383a-6b53-42fe-8c78-629661ff9d5c

### CODE REVIEW

File: src/components/AppNavHeader.tsx [hunk 1]
```diff
@@ -18,7 +19,8 @@ export function AppNavHeader({
   return (
     <Appbar.Header style={styles.style} elevated>
       {back ? (
-        <Appbar.BackAction
+        <Appbar.Action
+          icon={ChevronLeft}
```
Comment (bruce-glazier) [inline src/components/AppNavHeader.tsx:23]: 🙌

File: example/storybook/stories/TrackTile/util/MockEnvironmentDecorator.tsx [hunk 2]
```diff
@@ -6,13 +6,13 @@ import {
   Tracker,
   TrackTileServiceProvider,
   useAxiosTrackTileService,
-} from '../../../../src/components/TrackTile/main';
+} from '../../../../../src/components/TrackTile/main';
```
Comment (bruce-glazier) [inline example/storybook/stories/TrackTile/util/MockEnvironmentDecorator.tsx:9]: Opps. Thanks for fixing this, seems like there must be something we can do to allow an invalid path from being merged in.

**Comment by bruce-glazier** [comment]
I'm thinking that we should move the activity indicator below the parent post on the thread screen so the UI doesn't jump around so much.

**Comment by sternetj** [comment]
@bruce-glazier I updated the flat list to use two indicators. 1 for the the initial load which shows up beneath the post and another for the refresh control. Another approach might be to add a skeleton loader instead of the activity indicator but figured it would be better to wait on that.

https://github.com/lifeomic/react-native-sdk/assets/2295908/df63582e-6b06-489b-8273-579e5257e35f

File: src/screens/CircleThreadScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import { HomeStackScreenProps } from '../navigators/types';
+import { Thread } from '../components/Circles/Thread';
+
+export const CircleThreadScreen = ({
+  navigation,
+  route,
+}: HomeStackScreenProps<'Home/Circle/Thread'>) => {
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      title: `${route.params.post.author?.profile.displayName}'s Post`,
```
Comment (bruce-glazier) [inline src/screens/CircleThreadScreen.tsx:11]: Nit: Could use i18next for this title

**Comment by bruce-glazier** [comment]
> @bruce-glazier I updated the flat list to use two indicators. 1 for the the initial load which shows up beneath the post and another for the refresh control. Another approach might be to add a skeleton loader instead of the activity indicator but figured it would be better to wait on that.
> 
>  thread-loading.indicators.mov

Looks good to me!

**Review by bruce-glazier — APPROVED** [review]

**Pull request merged**